packages/tupaia-web/src/features/Map/MapOverlaySelector/MapTableModal.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import styled from 'styled-components';
+import DownloadIcon from '@material-ui/icons/GetApp';
+import MuiIconButton from '@material-ui/core/IconButton';
+import { FlexColumn } from '@tupaia/ui-components';
+import { Modal } from '../../../components';
+import { MapTable, MeasureData } from '@tupaia/ui-map-components';
+import { useParams } from 'react-router';
+import { useEntityAncestors, useMapOverlays, useEntitiesWithLocation } from '../../../api/queries';
+import { useMapOverlayReport } from '../utils';
+import { processMeasureData } from '../MapOverlays/processMeasureData';
+
+const Wrapper = styled(FlexColumn)`
+  justify-content: flex-start;
+`;
+
+const Title = styled.div.attrs({
+  variant: 'h2',
+})`
+  font-size: 1.25rem;
+  line-height: 1.4;
+  font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  margin-bottom: 1.5rem;
+`;
+
+const IconButton = styled(MuiIconButton)`
+  margin-bottom: 1.5rem;
+`;
+
+const TitleWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  margin-bottom: 3rem;
+  position: relative;
+  border-bottom: 1px solid rgba(255, 255, 255, 0.12);
+`;
+
+const useEntitiesByMeasureLevel = (measureLevel?: string) => {
+  const { projectCode, entityCode } = useParams();
+  const getSnakeCase = (measureLevel?: string) => {
+    return measureLevel
+      ?.split(/\.?(?=[A-Z])/)
+      .join('_')
+      .toLowerCase();
+  };
+
+  return useEntitiesWithLocation(
+    projectCode,
+    entityCode,
+    {
+      params: {
+        includeRoot: false,
+        filter: {
+          type: getSnakeCase(measureLevel),
+        },
+      },
+    },
+    { enabled: !!measureLevel },
+  );
+};
+
+export const MapTableModal = ({ mapModalOpen }: any) => {
+  const handleCloseModal = () => {
+    setMapModalOpen(false);
+  };
+
+  const { projectCode, entityCode } = useParams();
+  const { selectedOverlay } = useMapOverlays(projectCode, entityCode);
+  const { data: entitiesData } = useEntitiesByMeasureLevel(selectedOverlay?.measureLevel);
+  const { data: mapOverlayData } = useMapOverlayReport();
+
+  if (!entitiesData || !mapOverlayData) {
+    return null;
+  }
+
+  const processedMeasureData = processMeasureData({
```

Comment:
If you wanted to, to make some things reusable between `DataVisualsLayer` and `MapTableModal` you could turn some of it into a hook, like combine `useEntitiesByMeasureLevel` and `processedMeasureData` and handle the fetching etc of all the data. Then it could just return `processedMeasureData` and `serieses`, and anything else we directly need. This could go in  `Map/utils` folder, since it will only ever be used inside the `Map` folder